### ISSUE TITLE
Check if Transaction ranges are valid

### ISSUE THREAD

**Issue by karlfloersch** [plasma-group/plasma-utils#38]
**Is your feature request related to a problem? Please describe.**
Transactions which have ranges that overlap internally are considered invalid. It would be nice if the serialization library could check for transactions which are invalid in this way.

**Describe the solution you'd like**
The SignedTransaction could have a function `validate()` which checks that there are no internally overlapping ranges.

**Comment by kfichter** [comment]
Good call, this is an easy improvement. We’re already validating signatures so might as well validate the whole thing.

**Comment by kfichter** [comment]
Might be good to combine this with the addition of a utility function that determines if there’s any overlap between two sets of ranges. Ends up being useful all over the place.
